### ISSUE TITLE
Migrations can no longer be run from within a transaction

### ISSUE THREAD

**Issue by jeffkenney** [tgriesser/knex#3063]
# Environment

Knex version: 0.16.0+
Database + version: Postgres 9.6.12
OS: MacOS 10.13

Select applicable tempalate from below.
If issue is about oracledb support tag @ atiertant. For MSSql tag @ smorey2 .
Rest of dialects doesn't need tags.

# Bug

**1. Explain what kind of behaviour you are getting and how you think it should do**

Since version 0.16.0, migrations can no longer be run within a transaction. My team is currently using this behavior to stand up a database for running automated tests, and reverting to a clean database by rolling back the transaction after all testing is complete.

Minimal example:

```javascript
const conn = require('knex')(...);
conn.transaction((txn) => {
  txn.migrate.latest();
});
```

**2. Error message**

`Cannot set user params on a transaction - it can only inherit params from main knex instance`

**3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to mysql or postgresql, then single file example which initializes
   needed data and demonstrates the problem.**

Example (including docker postgres setup): https://github.com/jeffkenney/knex-txn-issue

**Comment by elhigu** [comment]
Thanks for the report.

This might be hard to fix since IIRC that user params is used to create clone of query builder, with disabling custom identifier wrappers to make sure that internal names for migration tables etc are consistent...

**Comment by kibertoad** [comment]
Fixed in 0.16.4-next2
